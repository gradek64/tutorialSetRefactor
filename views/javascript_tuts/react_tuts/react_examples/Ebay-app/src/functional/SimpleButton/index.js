import React from 'react';
import PropTypes from 'prop-types';

// custom
import './simpleButton.scss';


const SimpleButton = ({ label, eventHandler }) => (
  <input
    className="simple-button"
    type="button"
    onClick={eventHandler}
    value={label}
  />
);
export default SimpleButton;

SimpleButton.propTypes = {
  eventHandler: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
};
