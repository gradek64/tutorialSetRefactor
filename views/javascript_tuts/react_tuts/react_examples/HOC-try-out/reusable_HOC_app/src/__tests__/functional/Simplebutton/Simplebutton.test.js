import React from 'react';
import { shallow } from 'enzyme';

/*
  *@this next line is defined in a root/jest.configuration.js
  *@so no need to import library
*/
// import toJson from 'enzyme-to-json';

// component for snapshot test;
import SimpleButton from '../../../functional/Simplebutton';

test('renders correctly', () => {
  const wrapper = shallow(<SimpleButton label="label" action={() => {}} />);

  // that would be with import of enzyme-to-json bu we got that in configuration
  // expect(toJson(wrapper)).toMatchSnapshot();
  expect(wrapper).toMatchSnapshot();
});
