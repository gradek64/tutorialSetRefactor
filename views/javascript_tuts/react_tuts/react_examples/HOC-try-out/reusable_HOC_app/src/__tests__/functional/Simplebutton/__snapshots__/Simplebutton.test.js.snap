// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "borderColor": "green",
      "color": "green",
    }
  }
>
  label
</button>
`;
